backend/src/rss-feed/rss-feed.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import sources from './rss-feed-sources';
+import { RssFeed } from './rss-feed.entity';
+import { Repository } from 'typeorm';
+const Parser = require('rss-parser');
+const parser = new Parser();
+
+
+@Injectable()
+export class RssFeedService {
+
+    constructor(@InjectRepository(RssFeed)
+    private readonly rssFeedRepository: Repository<RssFeed>) {
+
+    }
+
+    async onModuleInit() {
+        await this.fetchAndSaveRssFeeds();
+    }
+
+    async fetchAndSaveRssFeeds() {
+        const news = await parser.parseURL(sources.news);
+        await this.rssFeedRepository.save(news.items);
+        const stories = await parser.parseURL(sources.stories);
```

Comment:
you can parse both news and stories simultaneously

`await Promise.all([ parse(news), parse(stories)])`

This is a blocking operation when you are waiting for data that doesnt relate to each other.

Reply:
Similar to when you persist these thing to database, but if you use transaction, they are wrapped in 1 call anyway

Reply:
Yes I agree. I will add that.